### PR TITLE
Closes #541

### LINKED ISSUE #541: Output file selection in GenericCodeInterface
--------
Issue Description
--------
##### What did you expect to see happen?
I should be able to tell RAVEN which output file(s) from my code that I want it to read results from.

##### What did you see instead?
RAVEN expects to rename the output file to a name of its own selection, which isn't convenient for all codes that use the GenericCodeInterface.

##### Do you have a suggested fix for the development team?
Remove the "out~" naming convention and instead allow the user to identify which output file to read.

This issue comes at the request of a user from a Jan 24, 2017 email thread to the RAVEN user group.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #541 

##### What are the significant changes in functionality due to this change request?
Added the possibility (with fargs node) to specify a fixed output file

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: doc/user_manual/existing_interfaces.tex [hunk 1]
```diff
@@ -76,6 +76,31 @@ \subsection{Generic Interface}
...
+
+Example:
+\begin{lstlisting}[style=XML]
+    <Code name="poly" subType="GenericCode">
+      <executable>GenericInterface/poly_inp.py</executable>
+      <inputExtentions>.one,.two</inputExtentions>
+      <clargs   type='prepend' arg='python'/>
+      <clargs   type='input'   arg='-i'  extension='.one'/>
+      <fileargs type='input'   arg='two' extension='.two'/>
+      <fileargs  type="output" arg="fixed_output.csv"/>
```
Comment (PaulTalbot-INL) [inline doc/user_manual/existing_interfaces.tex]: This has the same syntax as the fileargs wildcard replacement.  Maybe we could add a syntax specifically for named files?

Comment (alfoa) [reply doc/user_manual/existing_interfaces.tex]: Addressed comment

**Comment by PaulTalbot-INL** [comment]
Upon tests passing, my review says this is good to merge in.

**Comment by alfoa** [comment]
Merging.... Test passed

**Pull request merged**